File: Python/199. Binary Tree Right Side View.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        sol=[]
        def solution(root, sol, l):
            if root==None:
                return sol
        
            if len(sol)==l:
                sol.append(root.val)
        
            sol = solution(root.right, sol ,l + 1)
            sol = solution(root.left, sol ,l + 1)
            return sol

        sol = solution(root, sol, 0)
        return sol

File: Java/199. Binary Tree Right Side View.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> solution(TreeNode root, List<Integer> sol, int l){
        if (root==null){
            return sol;
        }
        if (sol.size()==l){
            sol.add(root.val);
        }
        sol = solution(root.right, sol ,l + 1);
        sol = solution(root.left, sol ,l + 1);
        return sol;
    }
    
    public List<Integer> rightSideView(TreeNode root) {
        List sol=new ArrayList<>();
        sol = solution(root, sol, 0);
        return sol;
    }
}
